### PR TITLE
Remove xabuild completely

### PR DESCRIPTION
**Pull request by grendello**

### CODE REVIEW

File: tests/Mono.Android-Tests/Runtime-Microsoft.Android.Sdk/Mono.Android.NET-Tests.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
   <PropertyGroup>
     <TargetFramework>$(DotNetAndroidTargetFramework)</TargetFramework>
-    <SupportedOSPlatformVersion>21</SupportedOSPlatformVersion>
+    <SupportedOSPlatformVersion>23</SupportedOSPlatformVersion>
```
Comment (pjcollins) [inline tests/Mono.Android-Tests/Runtime-Microsoft.Android.Sdk/Mono.Android.NET-Tests.csproj:8]: If we bump this I think we'll need to also adjust the nightly test matrix values:
https://github.com/xamarin/xamarin-android/blob/de9d6c87b6ccab5c7784bdafa7d03c65d21814da/build-tools/automation/azure-pipelines-nightly.yaml#L82

I know there was also some discussion about bumping our min supported version to 23 in .NET 9, has that decision officially been made yet?

File: tests/Mono.Android-Tests/Runtime-Microsoft.Android.Sdk/AndroidManifest.xml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 ﻿<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0" package="Mono.Android.NET_Tests">
-	<uses-sdk android:minSdkVersion="21" />
```
Comment (jpobst) [inline tests/Mono.Android-Tests/Runtime-Microsoft.Android.Sdk/AndroidManifest.xml:2]: Is this removal intentional?

Comment (grendello) [reply tests/Mono.Android-Tests/Runtime-Microsoft.Android.Sdk/AndroidManifest.xml:2]: Yep, IIRC we generate that line automatically now, based on the csproj settings (or defaults)

**Review by jpobst — APPROVED** [review]
